src/Internal/Payment/Transition.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Class Transition
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment;
+
+use WCPay\Internal\Payment\Change;
+
+/**
+ * A Transition object used for logging.
+ * Contains the details of a state transition during the payment process.
+ */
+class Transition {
+
+	/**
+	 * State of the transition.
+	 *
+	 * @var string|null
+	 */
+	private $from_state;
+
+	/**
+	 * State of the transition.
+	 *
+	 * @var string|null
+	 */
+	private $to_state;
+
+	/**
+	 * Changes made during the transition.
+	 *
+	 * @var Change[]
+	 */
+	private $changes;
+
+	/**
+	 * Timestamp of the transition.
+	 *
+	 * @var int
+	 */
+	private $timestamp;
+
+	/**
+	 * Constructs the class, with all the parameters.
+	 *
+	 * @param int         $timestamp Timestamp of the transition.
+	 * @param string|null $from_state State of the transition.
+	 * @param string|null $to_state State of the transition.
+	 * @param array       $changes Changes made during the transition.
```

Comment:
This type is not allowed for param but the PHPDoc allows this way so doing that is better. 

```suggestion
	 * @param Change[]       $changes Changes made during the transition.
```

Reply:
Done